.github/workflows/proxy-tests.yml

```diff
@@ -5,37 +5,67 @@ on:
       branches: [ main, release/**]
     push:
       branches: [ main, release/*]
-      tags: [ v* ]
+      tags: [ v*, v*.*.* ]
+
+defaults:
+  run:
+    shell: bash
+
+permissions:
+  contents: read
 
 jobs:
     proxy-tests:
-        runs-on: ubuntu-latest
+        name: Proxy Tests
+        runs-on: [self-hosted, Linux, medium, ephemeral]
         steps:
+            - name: Harden Runner
+              uses: step-security/harden-runner@63c24ba6bd7ba022e95695ff85de572c04a18142 # v2.7.0
+              with:
+                egress-policy: audit
+
             - name: Checkout repository
-              uses: actions/checkout@v3
+              uses: actions/checkout@1d96c772d19495a3b5c517cd2bc0cb401ea0529f # v4.1.3
 
-            - name: Install Lua
-              uses: leafo/gh-actions-lua@v8
-              with:
-                luaVersion: '5.3'
+            #############
+            # Note: leafo/gh-actions-lua doesn't seem to work with self-hosted runners
+            # See issue #33 https://github.com/leafo/gh-actions-lua/issues/33
+
+            #- name: Install Lua
+            #  uses: leafo/gh-actions-lua@35bcb06abec04ec87df82e08caa84d545348536e # v10.0.0
+            #  with:
+            #    luaVersion: '5.3'
+
+            #- name: Install LuaRocks
+            #  uses: leafo/gh-actions-luarocks@e65774a6386cb4f24e293dca7fc4ff89165b64c5 # v4.3.0
+            ##############
+
+            - name: Lua Install
+              run: |
+                sudo apt-get update
+                sudo apt-get install build-essential libreadline-dev lua5.3 liblua5.3-dev -y
 
-            - name: Install LuaRocks
-              uses: leafo/gh-actions-luarocks@v4
+            - name: Luarocks Install
+              run: |
+                sudo apt-get update && sudo apt-get install wget -y
+                wget https://luarocks.org/releases/luarocks-3.8.0.tar.gz
+                tar xzfp luarocks-3.8.0.tar.gz; cd luarocks-3.8.0
+                ./configure && make && sudo make install
 
             - name: Install lunatest
-              run: luarocks install lunatest
+              run: sudo luarocks install lunatest
```

Comment:
why sudo on all of these?

Reply:
When I ran it without sudo it failed permissions checks on the self-hosted runner.

Reply:
got it 👍 
tnxs!!